### PR TITLE
refactor: Making issuers & secret optional

### PR DESCRIPTION
**Pull request by lxfontes**
Continuation of https://github.com/wasmCloud/wasmcloud-operator/pull/67

Moving `secretName` and `issuers` to optional since they are no longer needed as of 1.0 https://github.com/wasmCloud/wasmCloud/commit/bc5d296f3a58bc5e8df0da7e0bf2624d03335d9f

We still mount the `secretName` for nats credentials when specified.

### CODE REVIEW

File: src/controller.rs [hunk 1]
```diff
@@ -498,24 +480,7 @@ fn pod_template(config: &WasmCloudHostConfig, _ctx: Arc<Context>) -> PodTemplate
...
-                Volume {
-                    name: "nats-creds".to_string(),
-                    secret: Some(SecretVolumeSource {
-                        secret_name: Some(config.spec.secret_name.clone()),
-                        ..Default::default()
-                    }),
-                    ..Default::default()
-                },
-            ]),
+            volumes: Some(volumes.clone()),
```
Comment (lxfontes) [inline src/controller.rs:483]: this `spec.Volumes` was getting overwritten in https://github.com/wasmCloud/wasmcloud-operator/blob/6d64c20b60a4821a4db7291e98b7f06882f06014/src/controller.rs#L494 
this change makes it consistent.

File: src/controller.rs [hunk 2]
```diff
@@ -121,31 +111,37 @@ async fn reconcile_crd(config: &WasmCloudHostConfig, ctx: Arc<Context>) -> Resul
...
-    let secret = secrets.get(&cfg.spec.secret_name).await.map_err(|e| {
-        warn!("Failed to read secrets: {}", e);
-        e
-    })?;
-    let s = Secrets::from_k8s_secret(&secret).map_err(|e| {
-        warn!("Failed to read secrets: {}", e);
-        Error::SecretError(format!(
-            "Failed to read all secrets from {}: {}",
-            secret.metadata.name.unwrap(),
+    let mut s = Secrets {
```
Comment (lxfontes) [inline src/controller.rs:114]: not a fan of `s` here, however tried to minimize touch point

File: crates/types/src/v1alpha1/wasmcloud_host_config.rs [hunk 3]
```diff
@@ -22,7 +22,7 @@ pub struct WasmCloudHostConfigSpec {
     pub host_replicas: u32,
     /// A list of cluster issuers to use when provisioning hosts. See
     /// https://wasmcloud.com/docs/deployment/security/zero-trust-invocations for more information.
-    pub issuers: Vec<String>,
+    pub issuers: Option<Vec<String>>,
```
Comment (joonas) [inline crates/types/src/v1alpha1/wasmcloud_host_config.rs:25]: We should probably update the comment above to note that this option is deprecated as of wasmCloud 1.0

File: crates/types/src/v1alpha1/wasmcloud_host_config.rs [hunk 4]
```diff
@@ -39,7 +39,7 @@ pub struct WasmCloudHostConfigSpec {
     pub nats_leaf_image: Option<String>,
     /// The name of a secret containing the primary cluster issuer key along with an optional set
     /// of NATS credentials.
-    pub secret_name: String,
+    pub secret_name: Option<String>,
```
Comment (joonas) [inline crates/types/src/v1alpha1/wasmcloud_host_config.rs:42]: I think it's worth updating the comment above to note that this secret is expected to contain a set of NATS Credentials if it's set

File: src/controller.rs (hunk 2 above)
Comment (joonas) [reply src/controller.rs:114]: I would rename this to `secrets` and call the line above ([#112](https://github.com/wasmCloud/wasmcloud-operator/pull/68/files#diff-1fd48ab302efbb91fe84507671007262db18094467e4ad707d036f60ff22e0eaR112)) `secrets_api` or something instead.

**Review by joonas — APPROVED** [review]
Overall this looks good to me 👍

File: src/controller.rs [hunk 5]
```diff
@@ -119,33 +109,39 @@ async fn reconcile_crd(config: &WasmCloudHostConfig, ctx: Arc<Context>) -> Resul
...
-    let secret = secrets.get(&cfg.spec.secret_name).await.map_err(|e| {
-        warn!("Failed to read secrets: {}", e);
-        e
-    })?;
-    let s = Secrets::from_k8s_secret(&secret).map_err(|e| {
-        warn!("Failed to read secrets: {}", e);
-        Error::SecretError(format!(
-            "Failed to read all secrets from {}: {}",
-            secret.metadata.name.unwrap(),
+    let mut secrets = Secrets {
```
Comment (protochron) [inline src/controller.rs:114]: `let mut secrets = Secrets.default()` is a better way to write this.

**Review by protochron — APPROVED** [review]

**Pull request merged**